openscap_report/scap_results_parser/parsers/info_of_test_parser.py

```diff
@@ -101,17 +102,27 @@ def get_object(self, id_object):
         object_dict["object_data"] = self._get_object_items(xml_object, xml_collected_object)
         return OvalObject(**object_dict)
 
+    def _transform_tuples_to_dict(self, array):
+        out_dict = {}
+        for key, value in array:
+            if key in out_dict:
+                key = self._get_unique_key(key)
+            out_dict[key] = value
+        array.clear()
+        array.append(out_dict)
+
     def _get_object_items(self, xml_object, xml_collected_object):
         out = []
         for element in xml_object.iterchildren():
             id_in_dict = self._get_unique_id_in_dict(element, out)
             if element.text and element.text.strip():
-                out.append({id_in_dict: element.text})
+                out.append((id_in_dict, element.text))
             else:
-                out.append({id_in_dict: self._get_ref_var(element, xml_collected_object)})
-        if len(out) > 1:
-            return [dict(pair for dict_item in out for pair in dict_item.items())]
-        return out
+                out.append((id_in_dict, self._get_ref_var(element, xml_collected_object)))
+        self._transform_tuples_to_dict(out)
+        if out:
+            return out
+        return None
```

Comment:
First of all you can just `return out or None`.

But I don't quite understand why you're trying to use None instead of an empty list if there are no elements. What benefit does it give? I can see a drawback, though. In situations when one would iterate over a list there always should be a check if the list is not None.

Reply:
Usage `None` instead empty array. I want to state that there is no data in the JSON for the given item. A dictionary filter could solve this.

Reply:
Sure, but why do you think that
```
foo {
   bars_list: None
}
```
is better than
```
foo {
   bars_list: []
}
```
?

Reply:
In terms of python code, it is better to use an empty array. As for the JSON content, I think it's better to use null because it's more obvious that there's no data or this data is optional. I think it is about a discussion about JSON content. Which is the best value to use for cases when there is no data?

Reply:
You're right, let's discuss it in the JSON structure issue. But here is some homework for you: https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/ :)